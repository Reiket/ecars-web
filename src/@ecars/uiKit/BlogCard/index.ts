import {BlogCardContent} from '@ecars/uiKit/BlogCard/components/BlogCardContent';
import {BlogCardComponent} from '@ecars/uiKit/BlogCard/BlogCardComponent';
import {BlogCardImage} from '@ecars/uiKit/BlogCard/components/BlogCardImage';

export const BlogCard = Object.assign(BlogCardComponent, {
  Content: BlogCardContent,
  Image: BlogCardImage,
});
