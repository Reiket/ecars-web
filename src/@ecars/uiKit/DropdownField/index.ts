import {DropdownFieldComponent} from '@ecars/uiKit/DropdownField/DropdownFieldComponent';
import {DropdownFieldWrapper} from '@ecars/uiKit/DropdownField/components/DropdownFieldWrapper';

export const DropdownField = Object.assign(DropdownFieldComponent, {
  Wrapper: DropdownFieldWrapper,
});
