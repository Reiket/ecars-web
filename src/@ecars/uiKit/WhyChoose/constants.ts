import {withBlockClass} from 'ecars-web-lib';
import {PageWrapper} from '@ecars/uiKit/Page/components/PageWrapper';

export const WhyChooseWrapperHOC = withBlockClass(PageWrapper, 'why-choose');

export interface WhyChooseItemType {
  title: string;
  text: string;
  id: number;
}

export const whyChooseItemConfig: WhyChooseItemType[] = [
  {
    id: 1,
    title: 'Some headline',
    text: 'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.',
  },
  {
    id: 2,
    title: 'Some headline',
    text: 'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.',
  },
  {
    id: 3,
    title: 'Some headline',
    text: 'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.',
  },
];
