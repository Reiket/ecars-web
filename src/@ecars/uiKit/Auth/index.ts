import {AuthLayout} from '@ecars/uiKit/Auth/components/AuthLayout';
import {AuthPageWrapperHOC} from '@ecars/uiKit/Auth/constants';

export const AuthPage = Object.assign(AuthLayout, {
  Wrapper: AuthPageWrapperHOC,
});