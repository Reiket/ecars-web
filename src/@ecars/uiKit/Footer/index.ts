import {FooterComponent} from '@ecars/uiKit/Footer/FooterComponent';
import {FooterCopyWrapperHOC, FooterWrapperHOC} from '@ecars/uiKit/Footer/constants';
import {FooterCopy} from '@ecars/uiKit/Footer/components/FooterCopy';
import {FooterCopyList} from '@ecars/uiKit/Footer/components/FooterCopyList';
import {FooterColumn} from '@ecars/uiKit/Footer/components/FooterColumn';
import {FooterLogoBlock} from '@ecars/uiKit/Footer/components/FooterLogoBlock';
import {FooterLinkList} from '@ecars/uiKit/Footer/components/FooterLinkList';

export const Footer = Object.assign(FooterComponent, {
  Wrapper: FooterWrapperHOC,
  Column: FooterColumn,
  LogoBlock: FooterLogoBlock,
  LinkList: FooterLinkList,
});

export const Copy = Object.assign(FooterCopy, {
  Wrapper: FooterCopyWrapperHOC,
  List: FooterCopyList,
});
