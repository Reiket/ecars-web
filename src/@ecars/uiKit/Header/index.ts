import {HeaderComponent} from '@ecars/uiKit/Header/HeaderComponent';
import {HeaderNavbarWrapperHOC, HeaderWrapperHOC} from '@ecars/uiKit/Header/constants';
import {HeaderNavbar} from '@ecars/uiKit/Header/components/HeaderNavbar';
import {HeaderUtils} from '@ecars/uiKit/Header/components/HeaderUtils';
import {HeaderBurger} from '@ecars/uiKit/Header/components/HeaderBurger';
import {HeaderNavbarList} from '@ecars/uiKit/Header/components/HeaderNavbarList';

export const Header = Object.assign(HeaderComponent, {
  Wrapper: HeaderWrapperHOC,
  Utils: HeaderUtils,
  Burger: HeaderBurger,
});

export const Navbar = Object.assign(HeaderNavbar, {
  Wrapper: HeaderNavbarWrapperHOC,
  List: HeaderNavbarList,
});
