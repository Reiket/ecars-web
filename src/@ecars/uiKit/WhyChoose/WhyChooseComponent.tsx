import type {FC} from 'react';
import {WhyChoose} from '@ecars/uiKit/WhyChoose';

export const WhyChooseComponent: FC = () => (
  <WhyChoose.Wrapper name="why-choose">
    <div className="why-choose__body">
      <div className="why-choose__image">
        <img
          src="/assets/images/why/01.png"
          alt="why_choose_image"
        />
      </div>
      <div className="why-choose__content">
        <h2 className="why-choose__title section-title">Why choose us</h2>
        <p className="why-choose__text">
          Quis blandit turpis cursus in hac. In hendrerit gravida rutrum quisque. Pellentesque habitant morbi tristique
          senectus et. Eget gravida cum sociis natoque. Pharetra diam sit amet nisl suscipit adipiscing bibendum.
        </p>
        <p className="why-choose__text">
          Porttitor massa id neque aliquam. In fermentum posuere urna nec. Rhoncus aenean vel elit scelerisque mauris
          pellentesque. Nullam ac tortor vitae purus faucibus ornare suspendisse sed nisi. Consequat id porta nibh
          venenatis cras sed.
        </p>
        <WhyChoose.List />
      </div>
    </div>
  </WhyChoose.Wrapper>
);
