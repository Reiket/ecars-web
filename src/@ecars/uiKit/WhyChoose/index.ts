import {WhyChooseWrapperHOC} from '@ecars/uiKit/WhyChoose/constants';
import {WhyChooseComponent} from '@ecars/uiKit/WhyChoose/WhyChooseComponent';
import {WhyChooseItem} from '@ecars/uiKit/WhyChoose/components/WhyChooseItem';
import {WhyChooseList} from '@ecars/uiKit/WhyChoose/components/WhyChooseList';

export const WhyChoose = Object.assign(WhyChooseComponent, {
  Wrapper: WhyChooseWrapperHOC,
  Item: WhyChooseItem,
  List: WhyChooseList,
});
