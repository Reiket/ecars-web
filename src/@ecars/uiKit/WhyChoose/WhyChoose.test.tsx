import {render} from '@testing-library/react';
import {WhyChoose} from '@ecars/uiKit/WhyChoose';

describe('WhyChoose Component', () => {
  test('render component correctly', () => {
    const {container} = render(<WhyChoose />);
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
