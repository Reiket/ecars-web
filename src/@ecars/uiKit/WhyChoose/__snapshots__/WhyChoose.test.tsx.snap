// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WhyChoose Component > render component correctly 1`] = `
<div>
  <div
    class="why-choose"
  >
    <section
      class="why-choose__section section"
    >
      <div
        class="why-choose__container container container--lg"
      >
        <div
          class="why-choose__body"
        >
          <div
            class="why-choose__image"
          >
            <img
              alt="why_choose_image"
              src="/assets/images/why/01.png"
            />
          </div>
          <div
            class="why-choose__content"
          >
            <h2
              class="why-choose__title section-title"
            >
              Why choose us
            </h2>
            <p
              class="why-choose__text"
            >
              Quis blandit turpis cursus in hac. In hendrerit gravida rutrum quisque. Pellentesque habitant morbi tristique senectus et. Eget gravida cum sociis natoque. Pharetra diam sit amet nisl suscipit adipiscing bibendum.
            </p>
            <p
              class="why-choose__text"
            >
              Porttitor massa id neque aliquam. In fermentum posuere urna nec. Rhoncus aenean vel elit scelerisque mauris pellentesque. Nullam ac tortor vitae purus faucibus ornare suspendisse sed nisi. Consequat id porta nibh venenatis cras sed.
            </p>
            <ul
              class="why-choose__why-choose-list why-choose-list"
            >
              <li
                class="why-choose-list__item"
              >
                <h4
                  class="why-choose-list__title title-why-choose-list"
                >
                  Some headline
                </h4>
                <p
                  class="why-choose-list__text"
                >
                  Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.
                </p>
              </li>
              <li
                class="why-choose-list__item"
              >
                <h4
                  class="why-choose-list__title title-why-choose-list"
                >
                  Some headline
                </h4>
                <p
                  class="why-choose-list__text"
                >
                  Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.
                </p>
              </li>
              <li
                class="why-choose-list__item"
              >
                <h4
                  class="why-choose-list__title title-why-choose-list"
                >
                  Some headline
                </h4>
                <p
                  class="why-choose-list__text"
                >
                  Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.
                </p>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
